applications/asset_tracker_v2/src/main.c

```diff
@@ -432,37 +376,15 @@ static void data_get(void)
 		app_module_event->data_list[count++] = APP_DATA_MODEM_STATIC;
 	}
 
-	if (!app_cfg.no_data.neighbor_cell) {
-		app_module_event->data_list[count++] = APP_DATA_NEIGHBOR_CELLS;
-		app_module_event->timeout = DATA_FETCH_TIMEOUT_NEIGHBORHOOD_SEARCH;
-	}
-
-	/* The reason for having at least 75 seconds sample timeout when
-	 * requesting GNSS data is that the GNSS module on the nRF9160 modem will always
-	 * search for at least 60 seconds for the first position fix after a reboot. This limit
-	 * is enforced in order to give the modem time to perform scheduled downloads of
-	 * A-GPS data from the GNSS satellites.
-	 *
-	 * However, if A-GPS data has been downloaded via the cloud connection and processed
-	 * before the initial GNSS search, the actual GNSS timeout set by the application is used.
-	 *
-	 * Processing A-GPS before requesting GNSS data is enabled by default,
-	 * and the time that the application will wait for A-GPS data before including GNSS
-	 * in sample requests can be adjusted via the
-	 * CONFIG_APP_REQUEST_GNSS_WAIT_FOR_AGPS_THRESHOLD_SEC Kconfig option. If this option is
-	 * set to -1, the application module will not request GNSS unless A-GPS data has been
-	 * processed.
-	 *
-	 * If A-GPS data has been processed the sample timeout can be ignored as the
-	 * GNSS will most likely time out before the sample timeout expires.
-	 *
-	 * When GNSS is requested, set the sample timeout to (GNSS timeout + 15 seconds)
-	 * to let the GNSS module finish ongoing searches before data is sent to cloud.
-	 */
+	if (IS_ENABLED(CONFIG_DATA_SAMPLE_GNSS_DEFAULT) ||
+	    IS_ENABLED(CONFIG_DATA_SAMPLE_NEIGHBOR_CELLS_DEFAULT)) {
```

Comment:
Location module checks the dynamic configuration coming from the cloud in `search_start()` function.

Reply:
I guess it would be better to check dynamic configuration here also instead of location module.
But do we have a chance it changes in between and I need to handle gnss and ncell being disabled in location module anyway or otherwise we may be sending location request without methods.

If we leave it only into location module, I think I need to add something there: https://github.com/nrfconnect/sdk-nrf/pull/8631/files#r1011384071

Reply:
You have to include `APP_DATA_LOCATION` in the `APP_EVT_DATA_GET` event, otherwise `search_start()` won't be even called in Location module, right?

Reply:
Yep, I guess I need the app_cfg.no_data check both here and in location module.

Reply:
Changed this now.